src/entities/models.ts

```diff
@@ -24,9 +24,13 @@ export interface DictionaryNum<T> {
 
 /**
  * @alpha
+ * why was this a class, not an interface?
  */
 export abstract class Dictionary<T> implements DictionaryNum<T> {
-  [id: string]: T | undefined
```

Comment:
FWIW, this definition came straight from `@ngrx/entity`.

I think the intent here is more that the user-level code understands that `booksState.entities["doesNotExist"]` returns `undefined`, vs a `Record<string, Book>` where it doesn't.

So, yeah, the internal casts are a bit weird, but it seems like an okay tradeoff to get the right behavior at the user level?

Reply:
Yeah. That's the lifelong discussion in the TS issues if an index signature should actually be undefined per default and it comes down to "if you really want to signal this, add the `| undefined`.
I guess in this place it makes sense, so I'll re-add it.

Reply:
And yeah, I never understood why this was an `abstract class` to begin with either